src/coresyftools/coresyftools/argument_parser.py

```diff
@@ -0,0 +1,108 @@
+from datetime import date
+import logging
+
+from argparse import ArgumentParser
+from manifest import validate_manifest, InvalidManifestException, MANIFEST_SCHEMA
+
+
+class CoReSyFArgumentParser():
+
+    MANIFEST_FILE_NAME = 'manifest.json'
+
+    def __init__(self, manifest, args=None, logger=None):
+        self.logger = logger or logging.getLogger(__name__)
+        self.bindings = {}
+        self.inputs = []
+        self.outputs = []
+        self.options = []
+        self.arg_parser = ArgumentParser()
+        self.manifest = manifest
+        validate_manifest(self.manifest)
+        is_valid, errors = self._validate_manifest(self.manifest)
+        if not is_valid:
+            raise InvalidManifestException(errors)
+
+    def _get_manifest_schema(self):
+        return MANIFEST_SCHEMA
+
+    def _validate_manifest(self, manifest):
+        return (True, [])
+
+    def _from_xsd_type_(self, xsd_type):
```

Comment:
1. Because it's a private method.
2. I have to say, now I know how `switch` works I prefer `if/elif`, since it is more declarative, once its semantic's is completely independent of the blocks it contains (what is not the case in `switch`, since the branches of executions can be different if the blocks contain `break` or not).
Furthermore, I don't see how the clean code recommendations can be applied here. 
a) Nothing said there implies that `switch` should be always used instead of `if/elif`;
b) The problems referred there are not present here.

Reply:
1. Private methods should have an underscore at the beggining of the method name, not at the end.
2. I was not saying that clean code recommended using `switch`, simply that if you decided to use switch you should see how they recommend using it. I think `switch` should be used when you have too many `if/elif` statements. But it's a matter of preference, so you can leave this how it is.

Reply:
1. Sorry, I was thinking in `_validate_manifest`.
2. My point is that is very easy to miss a `break` and then the `switch` doesn't have the `if/elif` semantic anymore, and using `if/elif` directly there isn't such risk. Still, you are right, is a matter of taste.